### ISSUE TITLE
PHP Fatal error after import csv with Chinese char.

### ISSUE THREAD

**Issue by NickWoo33** [nuxsmin/sysPass#533]
I click on the search, it show me:

There was an error
Internal Server Errorerror

Then I check the error_log, it shows:

PHP Fatal error:  Uncaught SP\\Core\\Exceptions\\SPException: [0]: Error de codificaci\xc3\xb3n (Malformed UTF-8 characters, possibly incorrectly encoded)\n\n  thrown in /var/www/html/syspass/inc/SP/Util/Json.class.php on line 71, referer: http://10.16.71.10/syspass/index.php

How can I fix it?

Thanks.

**Comment by NickWoo33** [comment]
# rpm -qa |grep php
php70w-mysqlnd-7.0.17-1.w7.x86_64
php70w-gd-7.0.17-1.w7.x86_64
php70w-opcache-7.0.17-1.w7.x86_64
php70w-common-7.0.17-1.w7.x86_64
php70w-mbstring-7.0.17-1.w7.x86_64
php70w-ldap-7.0.17-1.w7.x86_64
php70w-cli-7.0.17-1.w7.x86_64
php70w-mcrypt-7.0.17-1.w7.x86_64
php70w-xml-7.0.17-1.w7.x86_64
php70w-7.0.17-1.w7.x86_64
php70w-pdo-7.0.17-1.w7.x86_64
[root@cent7 httpd]# rpm -qa |grep json
json-c-0.11-4.el7_0.x86_64
json-glib-1.0.2-1.el7.x86_64

**Comment by nuxsmin** [comment]
Please take a look to this issue: http://stackoverflow.com/questions/16398041/php-json-encode-not-working-with-chinese-character-set#16398085

**Comment by nuxsmin** [comment]
TLDR; json_encode only works with UTF-8 encoding, so you would need to convert the CSV file before importing.

**Comment by NickWoo33** [comment]
Yes, i remove the it from database, and reimport with the utf-8, looks good now.

thanks.

**Comment by nuxsmin** [comment]
You're welcome

Thanks for the feedback!

**Comment by nuxsmin** [comment]
Umm, could try it out on the demo environment? (https:://demo.syspass.org)

So I can see what could be happening...

**Comment by NickWoo33** [comment]
I just logined the demo, and do the same testing, same error.

I create an account with Chinese Catagory and Customer, it's good to save the new account, but while do the search , it get error :+1: 

There was an error
OKerror

then the search page get stuck, need to refresh the page.

here's my test:

_测试
Testing in Chinese Char_

**Comment by nuxsmin** [comment]
Really weird, because some other app modules, like customers, uses the same functions without any issue.

I've done some tweaks to avoid the error, but the results are not shown instead.

Thanks for testing!, I'll try to find out the error.

**Comment by NickWoo33** [comment]
i had set the char to utf8 in mysql, still the same, :(

**Comment by NickWoo33** [comment]
So I just need to re-clone all the dir to update my site ? or could you show me how to update it in my site ?

Thanks.

**Comment by nuxsmin** [comment]
Hello, right, you can just replace the whole directory except the config one. Please take care about the directories permissions and owner.
